examples/pytorch/test_accelerate_examples.py

```diff
@@ -0,0 +1,317 @@
+# coding=utf-8
+# Copyright 2018 HuggingFace Inc..
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import json
+import logging
+import os
+import sys
+from unittest.mock import patch
+
+import torch
+
+from transformers.testing_utils import TestCasePlus, get_gpu_count, slow, torch_device
+from transformers.utils import is_apex_available
+
+
+SRC_DIRS = [
+    os.path.join(os.path.dirname(__file__), dirname)
+    for dirname in [
+        "text-generation",
+        "text-classification",
+        "token-classification",
+        "language-modeling",
+        "multiple-choice",
+        "question-answering",
+        "summarization",
+        "translation",
+        "image-classification",
+        "speech-recognition",
+        "audio-classification",
+        "speech-pretraining",
+        "image-pretraining",
+    ]
+]
+sys.path.extend(SRC_DIRS)
+
+
+if SRC_DIRS is not None:
+    # import run_audio_classification_no_trainer
+    import run_clm_no_trainer
+
+    # import run_generation_no_trainer
+    import run_glue_no_trainer
+
+    # import run_image_classification_no_trainer
+    # import run_mae_no_trainer
+    import run_mlm_no_trainer
+    import run_ner_no_trainer
+    import run_qa_no_trainer as run_squad_no_trainer
+
+    # import run_seq2seq_qa_no_trainer as run_squad_seq2seq
+    # import run_speech_recognition_ctc_no_trainer
+    # import run_speech_recognition_seq2seq_no_trainer
+    import run_summarization_no_trainer
+    import run_swag_no_trainer
+    import run_translation_no_trainer
+
+    # import run_wav2vec2_pretraining_no_trainer_no_trainer
+
+
+logging.basicConfig(level=logging.DEBUG)
+
+logger = logging.getLogger()
+
+
+def get_setup_file():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-f")
+    args = parser.parse_args()
+    return args.f
+
+
+def get_results(output_dir):
+    results = {}
+    path = os.path.join(output_dir, "all_results.json")
+    if os.path.exists(path):
+        with open(path, "r") as f:
+            results = json.load(f)
+    else:
+        raise ValueError(f"can't find {path}")
+    return results
+
+
+def is_cuda_and_apex_available():
+    is_using_cuda = torch.cuda.is_available() and torch_device == "cuda"
+    return is_using_cuda and is_apex_available()
+
+
+class ExamplesTests(TestCasePlus):
+    def test_run_glue(self):
```

Comment:
Nit but I would call them the same as the script (so all with `no_trainer` suffixes) just so it's obvious which script we are testing.